drivers/usbhost/hid_parser.c

```diff
@@ -274,8 +274,8 @@ int hid_parsereport(FAR const uint8_t *report, int rptlen,
             {
               collectionpath->usage.usage = usage[0];
 
-              for (i = 0; i < nusage; i++)
-                usage[i] = usage[i + 1];
+              for (i = 1; i < nusage; i++)
```

Comment:
I understand the code does the same thing as before, only differently. I don't understand by clang-check considers the other version problematic and this one does not.

Reply:
the original code reads usage[nusage+1]. the new one doesn't.

Reply:
got it